salt/payload.py

```diff
@@ -132,9 +133,11 @@ def loads(self, msg, encoding=None):
                 # Due to this, if we don't need it, don't pass it at all so
                 # that under Python 2 we can still work with older versions
                 # of msgpack.
-                return msgpack.loads(msg, use_list=True, encoding=encoding)
+                ret = msgpack.loads(msg, use_list=True, encoding=encoding)
             else:
-                return msgpack.loads(msg, use_list=True)
+                ret = msgpack.loads(msg, use_list=True)
+            if six.PY3 and not raw:
+                ret = salt.transport.frame.decode_embedded_strs(ret)
```

Comment:
I debated a lot about this, and went through all of the sale code and concluded that it would be better this way, the only problem I ran into was the file client which needs the data in bytes, all other data was operational and so I felt it was ok to assume all operational data can be decoded. Also, the only really large data sets are in the file client, so avoiding them there is also wise. And yes, I don't want to double decode, I just need to clean up some more.